src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -40,40 +41,38 @@
     jnp: Optional[ModuleType] = None  # type:ignore[no-redef]
 
 
-def _make_unary_array_field_intrinsic_func(builtin_name: str, array_builtin_name: str) -> Callable:
-    def _builtin_unary_op(a: _BaseNdArrayField) -> common.Field:
-        xp = a.__class__.array_ns
+def _make_builtin(builtin_name: str, array_builtin_name: str) -> Callable:
+    def _builtin_op(*fields: common.Field) -> common.Field:
```

Comment:
```suggestion
    def _builtin_op(*fields: common.Field | core_defs.SCALAR_TYPES) -> common.Field:
```
or something like
```suggestion
    def _builtin_op(self: common.Field, *fields: common.Field | core_defs.SCALAR_TYPES) -> common.Field:
```
Not sure if using self outside of a class is good practice, but afterall this will eventually be a method.

Reply:
not in call cases they will be methods